L1Trigger/L1TTwinMux/src/DTLowQMatching.cc

```diff
@@ -20,69 +20,73 @@
 using namespace std;
 
 DTLowQMatching::DTLowQMatching(L1MuDTChambPhContainer* inphiDTDigis, L1MuDTChambPhContainer inphiRPCDigis)  {
-  const std::vector<L1MuDTChambPhDigi> * vInCon=inphiDTDigis->getContainer();
-  m_phiDTDigis->setContainer( * vInCon);
- const std::vector<L1MuDTChambPhDigi> * vInCon2=inphiRPCDigis.getContainer();
- m_phiRPCDigis.setContainer(*vInCon2);
-
- m_phiRPCDigis2=inphiRPCDigis;
+m_phiDTDigis=inphiDTDigis;
+ m_phiRPCDigis=inphiRPCDigis;
 };
 
 
 void DTLowQMatching::run( const edm::EventSetup& c) {
 
-  const L1TwinMuxParamsRcd& tmParamsRcd = c.get<L1TwinMuxParamsRcd>();
+  const L1TTwinMuxParamsRcd& tmParamsRcd = c.get<L1TTwinMuxParamsRcd>();
   tmParamsRcd.get(tmParamsHandle);
-  const L1TwinMuxParams& tmParams = *tmParamsHandle.product();
+  const L1TTwinMuxParams& tmParams = *tmParamsHandle.product();
 
-	m_DphiWindow   = tmParams.get_DphiWindowBxShift();
+  m_DphiWindow   = tmParams.get_DphiWindowBxShift();
 
   Matching(0);
   Matching(1);
 }
 
 void DTLowQMatching::Matching(int track_seg){
 
-  L1MuDTChambPhDigi * dtts=0;
-  L1MuDTChambPhDigi * rpcts1=0;
-  
-
+  //L1MuDTChambPhDigi * dtts=0;
+  //L1MuDTChambPhDigi * rpcts1=0;
+  auto dtts=std::make_shared<L1MuDTChambPhDigi>(0,0,0,0,0,0,7,0,0,0);
+  auto rpcts1=std::make_shared<L1MuDTChambPhDigi>(0,0,0,0,0,0,7,0,0,0);
+  std::auto_ptr<L1MuTMChambPhContainer> m_phiRPCDigis_tm(new L1MuTMChambPhContainer);
+  const std::vector<L1MuDTChambPhDigi> *phiChambVector;
+  phiChambVector=m_phiRPCDigis.getContainer(); 
+  m_phiRPCDigis_tm->setContainer(*phiChambVector );
+  std::auto_ptr<L1MuTMChambPhContainer> m_phiDTDigis_tm(new L1MuTMChambPhContainer);
+  const std::vector<L1MuDTChambPhDigi> *phiChambVectorDT;
+  phiChambVectorDT=m_phiDTDigis->getContainer();
+  m_phiDTDigis_tm->setContainer(*phiChambVectorDT );
 
   int bx=0, wheel=0, sector=0, station=1;
    //cout<<"LowQ Matching  "<<track_seg<<endl;
-	for (wheel=-2;wheel<=2; wheel++ ){
-    for (sector=0;sector<12; sector++ ){
-     for (station=1; station<=4; station++){
-
-        for(bx=-3; bx<=3; bx++){
-        vector<int> delta_m, delta_p, delta_0;
-				  int matched = 0;
-          for(int rpcbx=bx-1; rpcbx<=bx+1; rpcbx++){
-            dtts=0; rpcts1=0; 
-            dtts = m_phiDTDigis->chPhiSegm(wheel,station,sector,bx ,track_seg);
+  for (wheel=-2;wheel<=2; wheel++ ){
+   for (sector=0;sector<12; sector++ ){
+    for (station=1; station<=4; station++){
+     for(bx=-3; bx<=3; bx++){
+         vector<int> delta_m, delta_p, delta_0;
+         int matched = 0;
+         for(int rpcbx=bx-1; rpcbx<=bx+1; rpcbx++){
+           // dtts=0; rpcts1=0; 
+           // dtts = m_phiDTDigis_tm->chPhiSegm(wheel,station,sector,bx ,track_seg);
+            dtts.reset(new L1MuDTChambPhDigi(0,0,0,0,0,0,7,0,0,0));
```

Comment:
This will probably give a bad performance since new L1MuDTChampPhDigis are constructed all the time.
Better to keep the raw pointers in this case.